api/content_query.go

```diff
@@ -36,25 +36,26 @@ const (
 
 var querySets map[string]int = map[string]int{
 	// search keys are done as lower case to make searches work regardless of case
-	spotlightStr:           querySpotlight,
-	articlesStr:            queryArticles,
-	bulletinsStr:           queryBulletins,
-	methodologiesStr:       queryMethodologies,
-	methodologyarticlesStr: queryMethodologyArticles,
-	staticdatasetsStr:      queryStaticDatasets,
-	timeseriesStr:          queryTimeseries,
+	spotlightStr:           QuerySpotlightFlag,
+	articlesStr:            QueryArticlesFlag,
+	bulletinsStr:           QueryBulletinsFlag,
+	methodologiesStr:       QueryMethodologiesFlag,
+	methodologyarticlesStr: QueryMethodologyArticlesFlag,
+	staticdatasetsStr:      QueryStaticDatasetsFlag,
+	timeseriesStr:          QueryTimeseriesFlag,
 
-	publicationsStr: queryArticles | queryBulletins | queryMethodologies | queryMethodologyArticles,
+	publicationsStr: QueryArticlesFlag | QueryBulletinsFlag | QueryMethodologiesFlag | QueryMethodologyArticlesFlag,
 
-	datasetsStr: queryStaticDatasets | queryTimeseries,
+	datasetsStr: QueryStaticDatasetsFlag | QueryTimeseriesFlag,
 }
 
 // getContentTypeParameter obtains a filter that defines a set of possible types
 func getContentTypeParameter(queryVars url.Values) int {
 	valArray, found := queryVars["type"]
 	if !found {
 		// no type specified, so return flags for all types
-		return querySpotlight | queryArticles | queryBulletins | queryMethodologies | queryMethodologyArticles | queryStaticDatasets | queryTimeseries
+		return QuerySpotlightFlag | QueryArticlesFlag | QueryBulletinsFlag | QueryMethodologiesFlag |
+			QueryMethodologyArticlesFlag | QueryStaticDatasetsFlag | QueryTimeseriesFlag
```

Comment:
I'm not familiar with this syntax - what does `int | int` do?

Reply:
Hi Dave,
Its "bitwise OR" and is a standard way of combining bit flags.
This should clarify:
https://stackoverflow.com/questions/28432398/difference-between-some-operators-golang